### ISSUE TITLE
Enable Dynamic Sampling

### ISSUE THREAD

**Issue by emmatyping** [getsentry/self-hosted#1866]
We should enable Dynamic Sampling!

Steps to accomplish this:
1. Figure out if we need to add any additional services (c.f https://github.com/getsentry/self-hosted/issues/1860#issuecomment-1351318749)
1. Switch it on in sentry.conf.example.py
1. Check with DS team that changes make sense
1. Note change in release notes

**Comment by pavels** [comment]
Just a small comment regarding this - https://github.com/getsentry/self-hosted/issues/1835#issuecomment-1349928636 - the breakage observed is with DS disabled. I have never tried to actually enable DS (`organizations:dynamic-sampling` feature flag).

**Comment by Agalin** [comment]
I've just enabled dynamic sampling in 22.11. That feature flag is not enough (even in 22.12 which removes server-side-sampling feature flag that's also necessary in 22.11).

Dynamic sampling needs a Quota instance with `get_blended_sample_rate` implemented and neither base Quota nor RedisQuota implements it. To my understanding this method provides "base" sampling for a project.

If it returns `None` (default for available implementations) then dynamic sampling won't generate any rules. Otherwise it'll generate a single uniform rule for `1.0` and a set of dynamic rules based on project settings for smaller values with returned one being used for the catch all rule.

I've added a custom implementation inheriting from RedisQuota that takes sampling rate from project options / org options / settings and it generates expected rules.

**Comment by petrprikryl** [comment]
Is there any time schedule for this according metrics changes in https://github.com/getsentry/self-hosted/releases/tag/23.9.1?

**Comment by hubertdeng123** [comment]
We don't have a set timeline for this, no.

**Comment by yanghua-ola** [comment]
Hi, is it correct to say that [Dynamic Sampling](https://develop.sentry.dev/application-architecture/dynamic-sampling/) is not yet available for self-hosted sentry?

Will https://github.com/getsentry/self-hosted/issues/2823 change anything?

**Comment by aldy505** [comment]
@shellmayr @ArthurKnaus Hey, do you know anything about this?

**Comment by shellmayr** [comment]
Hi @aldy505, apologies for the late response - sadly I have no additional information to provide here. Are you seeing specific issues when trying to enable dynamic sampling that I may be able to help with?

**Comment by aldy505** [comment]
> Hi [@aldy505](https://github.com/aldy505), apologies for the late response - sadly I have no additional information to provide here. Are you seeing specific issues when trying to enable dynamic sampling that I may be able to help with?

Hi @shellmayr, a few questions to make this clearer:
1. Do you think dynamic sampling should be enabled by default on self-hosted? If your answer is "no", then what are the implications?
2. If we are to enable dynamic sampling, do we need to deploy additional container (which means something like Snuba consumer, Sentry worker, Sentry processor, Sentry consumer) into the self-hosted stack?

**Comment by shellmayr** [comment]
Hi @aldy505, thanks for the clarifying questions!
1. Dynamic Sampling is a way to retain certain ("more valuable") data and reduce the overall amount of data ingested. I don't expect most self-hosted instances to ingest sufficient data for this to be actually worthwhile, but I also don't have the numbers so this kind of depends on the specific instance. There's some documentation on how this all works in our [develop documentation](https://develop.sentry.dev/application-architecture/dynamic-sampling/), if you're curious. To your question: If "enabled by default" means turning the feature flag on, I don't see why not - the behaviour of dynamic sampling with the default sample rate of 100% is equal to it being turned off, since you're retaining all of the data - so there should be no implications.
2. I'm not 100% familiar with the intricacies of self-hosted containers and how they work together, but in general the core functionality of Dynamic Sampling is part of Relay, so as long as there is a Relay running, and Relay communicates with the Sentry containers (as it generally needs to anyways), there should be no need to add additional containers to the self-hosted stack.
Hope this is useful!

**Comment by shellmayr** [comment]
I don't think that there are additional consumers for the dynamic sampling logic, as the Snuba consumers are at a later stage of the ingestion pipeline, and the entire sampling procedure is done in Relay.

**Comment by aldy505** [comment]
Okay. Thanks for the information! @shellmayr

We'll try to enable this on self-hosted by default.
